Source/Promise+Recover.swift

```diff
@@ -0,0 +1,62 @@
+//
+//  Promise+Recover.swift
+//  then
+//
+//  Created by Sacha Durand Saint Omer on 22/02/2017.
+//  Copyright © 2017 s4cha. All rights reserved.
+//
+
+import Foundation
+
+extension Promise {
+    
+    public func recover(with value: T) -> Promise<T> {
+        return Promise { resolve, _ in
+            self.then { t in
+                resolve(t)
+            }.onError { _ in
+                resolve(value)
+            }
+        }
+    }
+    
+    public func recover(_ errorType: Error, with value: T) -> Promise<T> {
+        // TODO Find a way to compare two Error types for equality
```

Comment:
We can try compare errors by their implementation of Equatable, otherwise fallback to comparision by _domain and _code. See https://github.com/Quick/Nimble/blob/db706fc1d7130f6ac96c56aaf0e635fa3217fe57/Sources/Nimble/Utils/Errors.swift#L37 as example